apps/gdal_footprint_lib.cpp

```diff
@@ -116,12 +117,177 @@ struct GDALFootprintOptions
     /*! Field name where to write the path of the raster. Empty if not desired */
     std::string osLocationFieldName = "location";
 
+    /*! Clears the osLocationFieldName var when set */
+    bool bClearLocation = false;
+
     /*! Whether to force writing absolute paths in location field. */
     bool bAbsolutePath = false;
 
     std::string osSrcNoData;
 };
 
+static std::unique_ptr<GDALArgumentParser> GDALFootprintAppOptionsGetParser(
+    GDALFootprintOptions *psOptions,
+    GDALFootprintOptionsForBinary *psOptionsForBinary)
+{
+    auto argParser = std::make_unique<GDALArgumentParser>(
+        "gdal_footprint", /* bForBinary=*/psOptionsForBinary != nullptr);
+
+    argParser->add_description(_("Compute footprint of a raster."));
+
+    argParser->add_epilog(_("For more details, consult "
+                            "https://gdal.org/programs/gdal_footprint.html"));
+
+    argParser->add_argument("-b")
+        .metavar("<band>")
+        .scan<'i', int>()
+        .append()
+        .store_into(psOptions->anBands)
+        .help(_("Band(s) of interest."));
+
+    argParser->add_argument("-combine_bands")
+        .choices("union", "intersection")
+        .action([psOptions](const auto &s)
+                { psOptions->bCombineBandsUnion = s == "union"; })
+        .default_value("union")
+        .help(_("Defines how the mask bands of the selected bands are combined "
+                "to generate a single mask band, before being vectorized."));
+
+    {
+        auto &group = argParser->add_mutually_exclusive_group();
+
+        group.add_argument("-ovr")
+            .metavar("<index>")
+            .scan<'i', int>()
+            .store_into(psOptions->nOvrIndex)
+            .help(_("Defines which overview level of source file must be used, "
+                    "when overviews are available on the source raster."));
+
+        group.add_argument("-srcnodata")
+            .metavar("<value>")
+            .nargs(nargs_pattern::at_least_one)
```

Comment:
I'm not too sure about that one. The current parser accepts things like ``-srcnodata "1 2"`` (the double quotes are actually removed by the shell, so in terms of argv[], this is 2 values: ``-srcnodata`` and ``1 2``). I understand that ``nargs(nargs_pattern::at_least_one)`` would allow to specify ``-srcnodata 1 2`` (which would be more natural), but is that stored correctly into osSrcNoData, which expects a space separator?

In gdalbuildvrt, I've just used the following to only support the syntax with values packed together in the same argument
```
    argParser->add_argument("-srcnodata")
        .metavar("\"<value>[ <value>]...\"")
        .store_into(psOptions->osSrcNoData);
```

Reply:
You are right, nargs_pattern::at_least_one or ::any leads to problems when the argument is the last one and there are positional arguments. I will use the same code as you did in gdalbuildvrt.

Reply:
done